build_tools/aws-sdk-code-generator/lib/aws-sdk-code-generator/views/rbs/client_class.rb

```diff
@@ -0,0 +1,133 @@
+          @aws_sdk_core_lib_path = options.fetch(:aws_sdk_core_lib_path)
+          @plugins = PluginList.new(options)
+          @waiters = AwsSdkCodeGenerator::RBS::Waiter.build_list(shape_dictionary: @shape_dictionary)
+        end
+
+        # @return [String|nil]
+        def generated_src_warning
+          GENERATED_SRC_WARNING
+        end
+
+        def service_name
+          @shape_dictionary.service_id
+        end
+
+        def client_option
+          plugin_options = documented_plugin_options(@plugins) + documented_plugin_options(@codegenerated_plugins)
+          build_keyword_arguments(plugin_options).join(", ")
+        end
+
+        def operations
+          @shape_dictionary.service.api["operations"].map do |key, body|
+            arguments = body.dig("input", "shape")&.then {
+              @shape_dictionary[_1].find(&:input?).as_keyword_arguments(from: :operations)
+            }
+            include_required = @shape_dictionary.service.api["shapes"][body.dig("input", "shape")]&.[]("required")&.empty?&.!
+            if AwsSdkCodeGenerator::Helper.operation_streaming?(body, @shape_dictionary.service.api)
+              block = " ?{ (*untyped) -> void }"
+            end
+            returns = body.dig("output", "shape")&.then { "Types::#{_1}" } || "::Aws::EmptyStructure"
+
+            method_name = Underscore.underscore(body.fetch("name"))
+            AwsSdkCodeGenerator::RBS::MethodSignature.new(
+              method_name: method_name,
+              overloads: [
+                "(#{arguments})#{block} -> #{returns}",
+                "(#{include_required ? "" : "?"}Hash[Symbol, untyped] params, ?Hash[Symbol, untyped] options)#{block} -> #{returns}",
+              ],
+            )
+          end
+        end
+
+        def waiters?
+          @waiters.size > 0
+        end
+
+        def waiters_first
+          [@waiters.first]
+        end
+
+        def waiters_others
+          @waiters[1..-1]
+        end
+
+        private
+
+        def documented_plugin_options(plugins)
+          i = 0
+          plugins.map(&:options).flatten.select(&:documented?).sort_by do |opt|
+            [opt.required ? 'a' : 'b', opt.name, i += 1]
+          end
+        end
+
+        def build_keyword_arguments(plugins)
+          buffer = plugins.map do |opt|
+            type = case opt.name
+              when :retry_mode
+                '("legacy" | "standard" | "adaptive")'
+              when :retry_jitter
+                "(:none | :equal | :full | ^(Integer) -> Integer)"
+              when :stub_responses, :endpoint, :credentials, :log_formatter, :client_side_monitoring_publisher, :logger, :endpoint_provider, :token_provider, :express_credentials_provider
+                # TODO: Just not supported yet
+                "untyped"
+              else
+                case opt.doc_type
+                when "Boolean"
+                  "bool"
+                when nil
+                  raise opt.inspect
```

Comment:
Whats the consequenece of using "untyped" here instead of raising?

I'm a little worried about this raising errors on new cases where we would just use "untyped".  (Also, if we are going to raise errors, we should include a more useful message).